### PR TITLE
Follow/unfollow transactions

### PR DESCRIPTION
**Pull request by megaherz**

### CODE REVIEW

File: cmd/decentrd/main.go [hunk 1]
```diff
@@ -58,6 +65,7 @@ func main() {
 	)
 	rootCmd.AddCommand(genutilcli.ValidateGenesisCmd(ctx, cdc, app.ModuleBasics))
 	rootCmd.AddCommand(AddGenesisAccountCmd(ctx, cdc, app.DefaultNodeHome, app.DefaultCLIHome))
+	rootCmd.AddCommand(AddGenesisCommunityModeratorsCmd(ctx, cdc, app.DefaultNodeHome, app.DefaultCLIHome))
```
Comment (mikluke) [inline cmd/decentrd/main.go:68]: I think we should somehow add this to gen-tx command

**Review by mikluke — COMMENTED** [review]
add follows to export/import genesis please

File: x/community/client/rest/rest.go [hunk 2]
```diff
@@ -34,6 +34,10 @@ func RegisterRoutes(cliCtx context.CLIContext, r *mux.Router, storeName string)
 	r.HandleFunc(fmt.Sprintf("/%s/likedPosts/{owner}", storeName), queryListUserLikedPostsHandler(cliCtx)).Methods(http.MethodGet)
 
 	r.HandleFunc(fmt.Sprintf("/%s/moderators", storeName), moderatorsHandler(cliCtx)).Methods(http.MethodGet)
+
+	r.HandleFunc(fmt.Sprintf("/%s/followers/follow/{whom}", storeName), createFollowHandler(cliCtx)).Methods(http.MethodPost)
+	r.HandleFunc(fmt.Sprintf("/%s/followers/unfollow/{whom}", storeName), createUnfollowHandler(cliCtx)).Methods(http.MethodPost)
+	r.HandleFunc(fmt.Sprintf("/%s/followers/{owner}/followees", storeName), queryFolloweesHandler(cliCtx)).Methods(http.MethodGet)
```
[the remark this answers is not in the document]
Comment (megaherz) [reply x/community/client/rest/rest.go:40]: This method is used to manually (in rest or cli) ensure that follow/unfollow works as expected. A list of followers/followee  will be implemented in Theseus.

File: x/community/genesis.go [hunk 3]
```diff
@@ -106,9 +141,16 @@ func ExportGenesis(ctx sdk.Context, k Keeper) GenesisState {
 		likes = append(likes, like)
 	}
 
+	var followers = make(map[string][]string)
+	k.IterateFollowers(ctx, func(who, whom sdk.Address) (stop bool) {
+		followers[who.String()] = append(followers[who.String()], whom.String())
+		return false
+	})
+
 	return GenesisState{
```
Comment (megaherz) [inline x/community/genesis.go:150]: @mikluke 
Exported genesis has the following structure
```json
    "community": {
      "followers": {
        "decentr1cc0n94re36kxvplma3g9wzjt59qteed4glhzf6": [
          "decentr1wg84j55z7zd2gx60wuq727qua7a0g9yp9dvy44"
        ],
        "decentr1wg84j55z7zd2gx60wuq727qua7a0g9yp9dvy44": [
          "decentr1cc0n94re36kxvplma3g9wzjt59qteed4glhzf6"
        ]
      },
      "likes": null,
      "moderators": [
        "decentr1cc0n94re36kxvplma3g9wzjt59qteed4glhzf6"
      ],
      "posts": null
    }
```

**Review by mikluke — APPROVED** [review]

**Pull request merged**